ducktape/cluster/node_container.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2017 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class NodeNotPresentError(Exception):
+    pass
+
+
+class InsufficientResourcesError(Exception):
+    pass
+
+
+class NodeContainer(object):
```

Comment:
Hmm.  I don't think it's a good fit for just a list or a set-- it stores nodes grouped by OS and possibly grouped in other ways in the future.  Also all the stuff discussed here: http://www.mit.edu/~6.005/fa14/classes/14-inheritance/ :)